### DISCUSSION TITLE
Divide a single graph into batches

### DISCUSSION THREAD

**Discussion by mhadnanali** [pyg-team/pytorch_geometric#5309] [Q&A]
Can we divide a single graph into batches any working example? E.G DBLP dataset I can not execute with 24 GB GPU {with GRACE, Deep Graph Contrastive Representation Learning } 

It is mentioned "PyG automatically takes care of batching multiple graphs into a single giant graph with the help of the [torch_geometric.loader.DataLoader](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.DataLoader) class." 

but what about one graph into mini-batches? is it feasible or not? if not then why (for my knowledge)

**Comment by EdisonLeeeee** [comment]
It is feasible to divide a single graph into batches. The simplest way is by [`NeighborLoader`](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.NeighborLoader), which allows mini-batch training on large-scale graphs where full-batch training is not feasible. Also, I think [`ClusterLoader`](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.ClusterLoader) is what you want to get different batches from a single graph.

**Comment by jiaqian** [comment]
I am wondering if batching step destroy a connected graph, say `G`,  into several batches? It will make some difference as running the whole graph `G` on one BIG batch? Does anyone have any idea?

**Comment by hahahaha90** [comment]
Hi, could you share the dblp dataset? thanks!

**Reply by rusty1s** [reply]
It is available at `torch_geometric.datasets.DBLP`.

**Reply by mhadnanali** [reply]
visit this page, you are going to find all of the datasets here. https://pytorch-geometric.readthedocs.io/en/latest/modules/datasets.html
